### ISSUE TITLE
Prometheus unable to fire alerts to Alertmanager

### ISSUE THREAD

**Issue by nc-ruth** [istio/istio#36082]
### Bug Description

We have had this issue for a really long time now, and we have tried multiple things for shortly mitigating the issue, but each mitigation eventually fails for us.

**Core issue:**
Prometheus is unable to fire alerts to Alertmanager. Prometheus is unable to hit specific pod IP's resolved from Alertmanagers headless service (`alertmanager-operated`). 

The error message from Prometheus when firing alerts is: 
`level=error ts=2021-11-15T13:51:10.269Z caller=notifier.go:527 component=notifier alertmanager=http://10.1.0.216:9093/api/v2/alerts count=26 msg="Error sending alert" err="bad response status 404 Not Found"` (referenced as **the error message** from now on)

**Prometheus versions used:** 
[kube-prometheus packaged by Bitnami
](https://github.com/bitnami/charts/tree/master/bitnami/kube-prometheus) which includes:
- Prometheus Operator (`version 0.51.2`)
- Prometheus (`version 2.28.1`)
- Alertmanager (`version 0.23.0`)

In Istio 1.9.8 , we had a sidecar configuration added to the Alertmanager, which mitigated the issue for a short while but eventually resulted in the error message: 
```
apiVersion: networking.istio.io/v1beta1
kind: Sidecar
metadata:
  namespace: monitoring
  name: alertmanager
spec:
  workloadSelector:
    labels:
      app: alertmanager
  ingress:
    - port:
        number: 9094
        protocol: TCP
        name: tcp-mesh
      defaultEndpoint: 0.0.0.0:9094
    - port:
        number: 9093
        protocol: TCP
        name: web
      defaultEndpoint: 0.0.0.0:9093
```
The above configuration also helped Alertmanager pods to go in in clustering mode, and find its "peers". 

In Istio 1.11.4 we did not have the need for such sidecar configuration, so we removed it. But after a few days, Prometheus started acting up, and in the logs of Prometheus it shows the error message.

We have multiple k8s clusters and multiple Prometheus instances, and eventually in all clusters all Prometheus instances are unable to find and fire alerts to Alertmanager, and in the logs of Prometheus it shows the error message.

If you restart a Prometheus instance, then that instance is able to find and fire alerts to Alertmanager, but after while it is unable to fire alerts to Alertmanager and in the logs of Prometheus it shows the error message.

We are really stuck with the issue, and have tried different versions of Istio, with all of them giving us the same issue where in the logs of Prometheus it shows the error message. 

We have tried the following versions of Istio:
- Istio 1.9.8
- Istio 1.11.4

Next thing we are going to try is to disable istio sidecar on Alertmanager, and see if this solves the issue. We are not super happy about this solution, since our wish is to run Istio mTLS strict mode in the near future. 

We have investigated the following threads with no success to our issues:
https://github.com/istio/istio/issues/33188 
https://github.com/prometheus-community/helm-charts/issues/190
https://github.com/prometheus-operator/prometheus-operator/pull/3904#issuecomment-897546236

### Version

```prose
client version: 1.11.4
control plane version: 1.11.4
data plane version: 1.11.4 (306 proxies), 1.9.8 (71 proxies)
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.4", GitCommit:"e87da0bd6e03ec3fea7933c4b5263d151aafd07c", GitTreeState:"clean", BuildDate:"2021-02-18T16:12:00Z", GoVers
ion:"go1.15.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.9", GitCommit:"ed93e7ea8f833824d05578765c2ad3efbebbfebe", GitTreeState:"clean", BuildDate:"2021-10-21T03:31:08Z", GoVers
ion:"go1.15.14", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by nc-ruth** [comment]
We just tried to disable istio-proxy (istio-inject = false) on alertmanager pods, and after awhile (21 hours), Prometheus stopped sending alerts to Alertmanager. This indicate that the problem is at Prometheus' istio sidecar.

I will try to debug further from the Prometheus sidecar, and see where the issue is.

**Comment by nc-ruth** [comment]
I just noticed the following issue: https://github.com/istio/istio/issues/34679 with the following PR https://github.com/istio/istio/pull/34680 which might solve this issue.

**Comment by nc-ruth** [comment]
We decided to wait till Istio 1.12 become stable, and then exclude port 9093 on Prometheus instances by using the following annotation:
```
      annotations:
        traffic.sidecar.istio.io/excludeOutboundPorts: "9093"
````

We are rolling Istio 1.12 RC1 out on a test cluster. Will update this issue as soon as we have tested whether Istio 1.12 solves our issue on our test cluster.

**Comment by dtu-ruth** [comment]
Hi @mjnagel, I am pretty sure that Istio 1.12 will solve the issue, I have not gotten to test it yet.

I might be able to test it tomorrow and let you know if it fixed it :)

**Comment by nc-ruth** [comment]
We are not seeing this issue anylonger.
I am closing the issue.

Thank you for all the inputs.

**Comment by kirankh7** [comment]
@nc-ruth, what fixed this issue?  istio 1.12

**Comment by nc-ruth** [comment]
Hi @kirankh7, I am pretty sure by upgrading to Istio 1.12, it solved the problem.
Please note that 1.12 is pretty outdated by now and is End Of Life (EOL), so I would advise to go for a higher version.
